src/components/AuthenticationForm/AuthenticationForm.tsx

```diff
@@ -0,0 +1,203 @@
+  const [backendError, setBackendError] = useState<{ [key: string]: string }>({
+    title: "",
+    message: "",
+  });
+  const hasBackendError = backendError.title !== "" && backendError.message !== "";
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  // Checks every inputError and if one has an error message, you can't signin/signup
+  const isMainButtonDisable = !Object.values(inputErrors).every(
+    (value: string) => value === correctState
+  );
+
+  // Update values of inputs and errors, according to the screen
+  useEffect(() => {
+    setInputErrors({});
+    authData[screen].forEach(({ id }) => {
+      setInputErrors((inputE) => ({ ...inputE, [id]: '' }));
+    });
+    setInputValues({});
+    authData[screen].forEach(({ id }) => {
+      setInputValues((value) => ({ ...value, [id]: '' }));
+    });
+  }, [screen]);
+
+  // Keep record of values and restart errors as a value changes
+  const onChangeHandler = (id: string, value: string) => {
+    setInputValues((storedValues) => ({ ...storedValues, [id]: value }));
+    setInputErrors((previousErrors) => ({ ...previousErrors, [id]: '' }));
+    // If password changes, passwordConfirmation validation status will change. Restart it
+    if (id === 'password')
+      setInputErrors((previousErrors) => ({
+        ...previousErrors,
+        passwordConfirmation: '',
+      }));
+  };
+
+  // Update errors according to rules
+  const onCheckRules = (id: string, value: string) => {
+    const rule = authRules.find((r) => r.id === id);
+    let validationResult = '';
+    if (rule) {
+      switch (id) {
+        case 'passwordConfirmation':
+          validationResult = rule.validate(value, inputValues.password);
+          break;
+        default:
+          validationResult = rule.validate(value);
+          break;
+      }
+      setInputErrors((previousErrors) => ({
+        ...previousErrors,
+        [id]: validationResult,
+      }));
+    }
+  };
+
+  const turnOffToast = () => {
+    setTimeout(() => {
+      setBackendError({
+        title: "",
+        message: ""
+      });
+    }, toastTime)
+  }
+
+  const submitForm = async (e: React.FormEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    try {
+      let user;
+
+      const studentRegex = /^[aA]0/g;
+      const userName = inputValues.email.split('@')[0];
+      if (screen === 'signIn') {
+        const { email, passwordLogin } = inputValues;
+        if (studentRegex.test(userName)) {
+          user = await logUser<StudentPromise>({
+            email: email.toLowerCase(),
+            password: passwordLogin,
+          })
+        } else {
+          user = await logUser<TeacherPromise>({
+            email: email.toLowerCase(),
+            password: passwordLogin,
+          })
+        }
+      } else {
+        const { firstName, lastName, email, password } = inputValues;
+
+        if (studentRegex.test(userName)) {
+          user = await createUser<StudentPromise>({
```

Comment:
These Endpoints are not longer useful. The backend end changed them. Make sure to update them